### ISSUE TITLE
Measure CPU/memory usage of child processes

### ISSUE THREAD

**Issue by ekisu** [F1bonacc1/process-compose#289]
## Feature Request
Right now, the CPU/memory usage displayed on the TUI only considers the spawned process, but does not consider the usage of their spawned subprocesses. It would be nice if the combined CPU/memory usage was shown instead.

#### Use Case:
In my case, I use [just](https://github.com/casey/just) to run the actual processes I need. I could imagine this also being useful for some servers that spawn subprocesses eg. nginx

#### Proposed Change:
Sum the usages of the spawned process and all its subprocesses; and display those on the TUI

#### Who Benefits From The Change(s)?
--

#### Alternative Approaches
Aside from looking at htop/similar tools, none really

**Comment by garder500** [comment]
I'm pushing this because i would need it too

**Comment by F1bonacc1** [comment]
Will be part of the next release.

**Comment by F1bonacc1** [comment]
This feature is now available in v1.63.0.
Please let me know if it works as expected.

**Comment by mythmon** [comment]
I don't think this is working as expected, or it has regressed. I'm running Version 1.73.0 and process-compose seems to only be showing the memory usage of the parent process. Here is a screenshot of process-compose and btop side-by-side showing memory usage. btop shows that the total process tree is using about 339 MB of memory, but process-compose only shows the 68 MB used by the parent process.

<img width="840" height="412" alt="Image" src="https://github.com/user-attachments/assets/c894b629-c4eb-4d49-b906-efcf9ad5eacf" />

**Comment by garder500** [comment]
> I don't think this is working as expected, or it has regressed. I'm running Version 1.73.0 and process-compose seems to only be showing the memory usage of the parent process. Here is a screenshot of process-compose and btop side-by-side showing memory usage. btop shows that the total process tree is using about 339 MB of memory, but process-compose only shows the 68 MB used by the parent process.
> 
> <img width="840" height="412" alt="Image" src="https://github.com/user-attachments/assets/c894b629-c4eb-4d49-b906-efcf9ad5eacf" />

I can confirm it regressed, it was working perfectly fine like 6 month ago, but it seem to returned to his original state before the "actual fix"

**Comment by F1bonacc1** [comment]
Unfortunately, I had to revert the recursive metrics collection because of the high CPU usage when there are many processes (above 30).
To re-enable it, start PC with the `--recursive-metrics` flag.
You are welcome to follow and comment on the upstream bug that I opened:
https://github.com/shirou/gopsutil/issues/1874

**Comment by mythmon** [comment]
Thanks! That's better than a regression. I'll follow along in the other issue.

**Comment by F1bonacc1** [comment]
Hi,

I added preprocessing optimizations in v1.85.0. The recursive metrics performance should be significantly better in this release.
I will wait for feedback and if the improvement is good enough for all the use cases (10s or even 100s of processes) I will make this the default behavior again.
